pkg/machine/applehv/machine.go

```diff
@@ -1111,7 +1119,38 @@ WantedBy=multi-user.target`
 			Name:     fmt.Sprintf("%s.mount", mnt.Tag),
 			Contents: machine.StrToPtr(fmt.Sprintf(mountUnit, mnt.Tag, mnt.Target)),
 		}
-		unitFiles = append(unitFiles, virtiofsAutomount, virtiofsMount)
+
+		// This "unit" simulates something like systemctl enable virtiofs-mount-prepare@
+		enablePrep := machine.Unit{
+			Enabled: machine.BoolToPtr(true),
+			Name:    fmt.Sprintf("virtiofs-mount-prepare@%s.service", mnt.Tag),
+		}
+
+		unitFiles = append(unitFiles, virtiofsAutomount, virtiofsMount, enablePrep)
 	}
+
+	// mount prep is a way to workaround the FCOS limitation of creating directories
+	// at the rootfs / and then mounting to them.
+	mountPrep := `
+[Unit]
+Description=Allow virtios to mount to /
+DefaultDependencies=no
+ConditionPathExists=!%f
+
+[Service]
+Type=oneshot
+ExecStartPre=chattr -i /
+ExecStart=mkdir -p '%f'
+ExecStopPost=chattr +i /
```

Comment:
Isn't this racy? While doing this with one unit is fine you do it in x units at the same time because systemd will start the units at the same time and systemd start in parallel by default unless explicit dependencies are configured.

Therefore this order is not safe at all. We could end up with something like this:
|unit 1| unit 2|
|-------|--------|
| chattr -i / | chattr -i /|
| mkdir | |
| chattr +i / | |
| | mkdir <-- this will fail  |
| | chattr +i / |

I think it would be much safer to do it in one unit with one `chattr -i /` then mkdir all you volumes and then one `chattr +i /` at the end, this would remove all races.

Reply:
Good point.

Reply:
I agree this *may* expose a race condition as systemd could run all the units in parallel. If this did expose a race condition and we knew the paths up front then  just doing it in a single unit would certainly prevent the race.

Reply:
@Luap99 is right it's racy